backtracepython/attributes/linux_memory_attribute_provider.py

```diff
@@ -0,0 +1,97 @@
+import platform
+import re
+import subprocess
+
+from backtracepython.attributes.attribute_provider import AttributeProvider
+
+
+class LinuxMemoryAttributeProvider(AttributeProvider):
+    is_linux = platform.system() == "Linux"
+    memory_match = re.compile(r'^(.+):\s+(\d+)\s*(.+)?$')
+
+    memory_parser = {
+        'MemTotal': 'system.memory.total',
+        'MemFree': 'system.memory.free',
+        'MemAvailable': 'system.memory.available',
+        'Buffers': 'system.memory.buffers',
+        'Cached': 'system.memory.cached',
+        'SwapCached': 'system.memory.swap.cached',
+        'Active': 'system.memory.active',
+        'Inactive': 'system.memory.inactive',
+        'SwapTotal': 'system.memory.swap.total',
+        'SwapFree': 'system.memory.swap.free',
+        'Dirty': 'system.memory.dirty',
+        'Writeback': 'system.memory.writeback',
+        'Slab': 'system.memory.slab',
+        'VmallocTotal': 'system.memory.vmalloc.total',
+        'VmallocUsed': 'system.memory.vmalloc.used',
+        'VmallocChunk': 'system.memory.vmalloc.chunk',
+    }
+
+
+    process_parser = {
+        'nonvoluntary_ctxt_switches': {
+            'parse': 'int',
+            'attr': 'sched.cs.involuntary',
+        },
+        'voluntary_ctxt_switches': {
+            'parse': 'int',
+            'attr': 'sched.cs.voluntary',
+        },
+        'FDSize': { 'parse': 'int', 'attr': 'descriptor.count' },
+        'FDSize': { 'parse': 'int', 'attr': 'descriptor.count' },
+        'VmData': { 'parse': 'kb', 'attr': 'vm.data.size' },
+        'VmLck': {  'parse': 'kb', 'attr': 'vm.locked.size' },
+        'VmPTE': { 'parse': 'kb', 'attr': 'vm.pte.size' },
+        'VmHWM': { 'parse': 'kb', 'attr': 'vm.rss.peak' },
+        'VmRSS': { 'parse': 'kb', 'attr': 'vm.rss.size' },
+        'VmLib': { 'parse': 'kb', 'attr': 'vm.shared.size' },
+        'VmStk': {'parse': 'kb', 'attr': 'vm.stack.size' },
+        'VmSwap': {'parse': 'kb', 'attr': 'vm.swap.size' },
+        'VmPeak': {'parse': 'kb', 'attr': 'vm.vma.peak' },
+        'VmSize': { 'parse': 'kb', 'attr': 'vm.vma.size' },
+    }
+
+    def get(self): 
+        result = {}
+        if not self.is_linux:
+            return result
```

Comment:
why not add this provider depending on that condition, instead of using conditions inside it?

Reply:
done